### PR TITLE
Event history UX update

### LINKED ISSUE #495: Missing Key in the en.json file?
This was brought up by @diarmidmackenzie in an issue in another case https://github.com/tripleblindmarket/covid-safe-paths/issues/483

On the Exposure History page, there is some English text that goes like this:
What does this mean? Individuals who don't exhibit symptoms can sometimes still carry the infection and infect others. Being careful about...

This text is not getting translated as it is not included in the en.json file. Is there a missing key for this?

### PR DESCRIPTION
**Pull request by tstirrat**
_Sorry for the size of this PR, doing the Theming blew it out. I am willing to split theme stuff into a separate PR if needed._

- Replaces moment js with dayjs (immutable = way less potential for mutation/state bugs)
- Added emotion's styled components, for cleaner styles, and ability to create a powerful theming system
- Developed a set of themes and a `<Theme>` component with these known themes
  - main (settings screen)
  - dark (when exposure is found, white on gray)
  - inverted (main screen, white on violet)

- Added a Typography component that encodes the design system from Upstatement, and is theme color aware
  - Currently this is not migrated to be used outside this PR, but I do plan to migrate theming and update other components (like Button) after this PR
- Tests for Typography, Theme,  and places with date math

Screenshots:

Screen Shot 2020-04-13 at 2 22 17 PM

Screen Shot 2020-04-13 at 2 22 41 PM

Fixes #495

### CODE REVIEW

**Comment by tstirrat** [comment]
Screen Shot 2020-04-11 at 5 24 41 PM

**Comment by tstirrat** [comment]
Added legend: 
Screen Shot 2020-04-13 at 12 13 36 PM

File: app/views/ExposureHistory/ExposureHistory.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+export const ExposureHistoryScreen = ({ navigation }) => {
+  const [history, setHistory] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchData() {
+      let dayBins = await GetStoreData(CROSSED_PATHS);
+
+      // dayBins = generateFakeIntersections(6); // handy for creating faux data
+
```
Comment (kenpugsley) [inline app/views/ExposureHistory/ExposureHistory.js:26]: I'm guessing some fixup in logic right about here.  I was able to do this in both Android and iOS by setting locations (so at least 1 location was set), picking the test authority, and then reloading the app.  The crossed results array looks like this 
`Crossing results:  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]`

Screen Shot 2020-04-13 at 5 54 26 PM

Comment (tstirrat) [reply app/views/ExposureHistory/ExposureHistory.js:26]: It only shows 14 days despite what's in dayBins

I've updated it to only show Possible exposure (charcoal bg) if there was an actual hit

And updated it to show orange on current day circle if there's an exposure

File: app/locales/languages.js [hunk 2]
```diff
@@ -18,6 +23,14 @@ import it from './it.json';
 // 5. REMOVE all empty translations. e.g. "key": "", this will allow fallback to the default: English
 // 6. import xyIndex from `./xy.json` and add the language to the block at the bottom
 
+const deviceLocale =
+  Platform.OS === 'ios'
+    ? NativeModules.SettingsManager.settings.AppleLocale ||
+      NativeModules.SettingsManager.settings.AppleLanguages[0] // iOS 13
+    : NativeModules.I18nManager.localeIdentifier;
+
```
[the remark this answers is not in the document]
Comment (tstirrat) [reply app/locales/languages.js:34]: Done.

**Comment by tstirrat** [comment]
Updated to show orange, green on today (if exposure minutes is >= 0), otherwise, show "today" color:

Screen Shot 2020-04-13 at 8 01 34 PM

**Comment by tstirrat** [comment]
Updated so that 0 bins do not count as exposure.

**Comment by tstirrat** [comment]
Fixes #495

**Pull request merged**